### PR TITLE
Update ctypes stubs for Python3.12

### PR DESCRIPTION
**Pull request by sobolevn**
Source: https://github.com/python/cpython/blob/3.12/Lib/ctypes/__init__.py#L572-L577

Basically, it depends on `SIZEOF_TIME_T` value to set a proper alias type.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
Stubtest complains it's not a union at runtime. (There are also a bunch of other stubtest complains, maybe it picked up the new beta?)

**Comment by sobolevn** [comment]
@JelleZijlstra you are right:
1. I've sent https://github.com/python/typeshed/pull/10291 to fix the CI for the new beta
2. It is not a `Union` on any concrete system, but it is a "meta-union", because on different systems it will be a different thing. So, I guess it is safe to allowlist this type

File: stdlib/ctypes/__init__.pyi [hunk 1]
```diff
@@ -181,6 +181,9 @@ class c_bool(_SimpleCData[bool]):
 if sys.platform == "win32":
     class HRESULT(_SimpleCData[int]): ...  # TODO undocumented
 
+if sys.version_info >= (3, 12):
+    c_time_t: TypeAlias = c_int32 | c_int64  # noqa: Y042
```
Comment (AlexWaygood) [inline stdlib/ctypes/__init__.pyi:185]: Unless this is literally an instance of `types.UnionType` at runtime, wouldn't it be better to do something like

```suggestion
    c_time_t: type[c_int32 | c_int64]
```

?

**Comment by AlexWaygood** [comment]
you now need to remove the unused allowlist entry for `c_time_t`

**Comment by AlexWaygood** [comment]
Thanks!

**Pull request merged**